include/aor.h

```diff
@@ -0,0 +1,436 @@
+
+  /// Removes any binding that had the given ID.  If there is no such binding,
+  /// does nothing.
+  void remove_binding(const std::string& binding_id);
+
+  /// Retrieve a subscription by To tag, creating an empty one if necessary.
+  Subscription* get_subscription(const std::string& to_tag);
+
+  /// Remove a subscription for the specified To tag.  If there is no
+  /// corresponding subscription does nothing.
+  void remove_subscription(const std::string& to_tag);
+
+  // Remove the bindings from an AOR object
+  void clear_bindings();
+
+  /// Retrieve all the bindings.
+  inline const Bindings& bindings() const { return _bindings; }
+
+  /// Retrieve all the subscriptions.
+  inline const Subscriptions& subscriptions() const { return _subscriptions; }
+
+  // Return the number of bindings in the AoR.
+  inline uint32_t get_bindings_count() const { return _bindings.size(); }
+
+  // Return the number of subscriptions in the AoR.
+  inline uint32_t get_subscriptions_count() const { return _subscriptions.size(); }
+
+  // Return the expiry time of the binding or subscription due to expire next.
+  int get_next_expires();
+
+  /// get_last_expires
+  ///
+  /// This returns the expiry time of the binding or subscription due to expire
+  /// last.
+  /// The expiry time is relative, so if this was called on an AoR containing a
+  /// single binding due to expire in 1 minute it would return 60.
+  /// This can be called on a empty AoR, where it will return 0.
+  ///
+  /// @return Return the expiry time of the binding or subscription due to
+  ///         expire last.
+  int get_last_expires() const;
+
+  /// Copy all site agnostic values from one AoR to this AoR. This copies basically
+  /// everything, but importantly not the CAS. It doesn't remove any bindings
+  /// or subscriptions that may have been in the existing AoR but not in the copied
+  /// AoR.
+  ///
+  /// @param source_aor           Source AoR for the copy
+  void copy_aor(const AoR& source_aor);
+
+  /// Patch an AoR with a partial update. The update covers adding or removing
+  /// individual bindings or subscriptions, replacing the AssociatedURIs,
+  /// incrementing the CSeq and setting the CSeq to at least a minimum value.
+  /// Any combination of the above is supported. This method can't be used to
+  /// change the timer ID or S-CSCF uri of the AoR.
+  ///
+  /// @param po PatchObject to patch the AoR with
+  void patch_aor(const PatchObject& po);
+
+  /// CSeq value for event notifications for this AoR.  This is initialised
+  /// to one when the AoR record is first set up and incremented every time
+  /// the record is updated while there are active subscriptions.  (It is
+  /// sufficient to use the same CSeq for each NOTIFY sent on each active
+  /// because there is no requirement that the first NOTIFY in a dialog has
+  /// CSeq=1, and once a subscription dialog is established it should
+  /// receive every NOTIFY for the AoR.)
```

Comment:
I do not understand this sentence in brackets. My understanding broke down on the word "active" FTR.

Reply:
Oh I think it's missing the word subscription; each active subscription

Reply:
Maybe I wasn't supposed to be reviewing this